mixnet/src/crypto.rs

```diff
@@ -0,0 +1,66 @@
+use std::{fmt::Debug, ops::Deref};
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// Private key for decrypting Sphinx packets.
+///
+/// The key type is basically X25519, but with small modifications: https://github.com/nymtech/sphinx/blob/ca107d94360cdf8bbfbdb12fe5320ed74f80e40c/src/crypto/keys.rs#L15
+pub struct PrivateKey(sphinx_packet::crypto::PrivateKey);
+
+impl PrivateKey {
+    /// Creates a random [`PrivateKey`].
+    pub fn new() -> Self {
+        Self(sphinx_packet::crypto::PrivateKey::new())
+    }
+}
+
+impl Clone for PrivateKey {
+    fn clone(&self) -> Self {
+        Self(sphinx_packet::crypto::PrivateKey::from(self.0.to_bytes()))
+    }
+}
```

Comment:
Do we really need this? One of the reasons of not having clone for private keys is to actually have a single source of responsibility for them. It helps on not confusing them or leaking them. That is why sphinx crate probably do not implement it itself.

Reply:
I added `Clone` because the [`NetworkBackend::Settings`](https://github.com/logos-co/nomos-node/blob/2d715ea031860386c46c1654bf3c724f6b3de4b0/nomos-services/network/src/backends/mod.rs#L15) requires `Clone`. The `PrivateKey` will be included in the [MixNodeConfig](https://github.com/logos-co/nomos-node/blob/2d715ea031860386c46c1654bf3c724f6b3de4b0/mixnet/src/node.rs#L12) that will be included in the `MixnetNetworkBackendSettings`. Let me check if we can make `Settings` not require `Clone`.

Reply:
It seems that it's not easy to remove Clone from the Settings type because of https://github.com/logos-co/Overwatch/blob/2f708060f78128b1236d892c87a2138ed0e56770/overwatch-rs/src/services/settings.rs#L25-L25, which may not be easy to change right now.

IMHO, it would be easier to leave the Clone in this PR as it is and find a better way to handle private keys for all of mixnet, consensus, and so on.

Reply:
> You can avoid the clone on the private key if in the settings you just accept raw bytes and then is validated inside.

Yeah, this is how the consensus service does, but then what's the point of not having the Clone on the private key if the raw bytes can be cloned freely and the private key can be reconstructed from it?

Reply:
> > You can avoid the clone on the private key if in the settings you just accept raw bytes and then is validated inside.
> 
> Yeah, this is how the consensus service does, but then what's the point of not having the Clone on the private key if the raw bytes can be cloned freely and the private key can be reconstructed from it? Maybe just for minimizing the chance of cloning the key?

Yeah, actually the right approach would be to select the file or stream from where to read the key in the settings instead of the raw bytes themselves. And that is wrong as well in consensus.

Reply:
Right. then, I'll use raw bytes here instead of explicitly implementing Clone. That's at least better until we have a sophisticated way

Reply:
Removed Clone: 899d898. Raw bytes will be added to the config at the next PR.